### PR TITLE
Application Connector Documentation: one click integration

### PR DESCRIPTION
**Pull request by akgalwas**
**Description**

Changes proposed in this pull request:

- One Click Integration description enhanced

**Related issue(s)**

#3441

### CODE REVIEW

File: docs/application-connector/docs/02-02-connector-service.md [hunk 1]
```diff
@@ -13,8 +13,19 @@ This diagram illustrates the client certificate generation flow in details:
...
+    - the URL of the Application Registry API
+    - the URL of the Event Service API
+    - the certificate renewal URL that should be used to rotate certificate
+    - the certificate revocation URL that should be used to revoke compromised certificates
+    - information uniquely identyfying the certificate (Application name)
+    - information required to generate a CSR  
+
+>**NOTE:** The external application should not hardcode any URLs. The information returned from metadata endpoint should be stored by the external application along with the certificate. This approach implicates less coupling and offers a great deal of flexibility. 
+
+>**NOTE:**  The external application can call the metadata endpoint to fetch information required to generate a CSR prior to certificate renewal. This approach makes certificate rotation process convenient and flexible, since the external application does not need to store information required to generate a CSR in its data model.
```
Comment (Szymongib) [inline docs/application-connector/docs/02-02-connector-service.md]: What do you think about converting this note to some kind of `Connectivity guidelines` section, which would contain such information?

File: docs/application-connector/docs/08-02-get-client-certificate.md [hunk 2]
```diff
@@ -75,6 +75,8 @@ A successful call returns the following response:
 }
 ```
 
+> **NOTE:** The response contains URLs to Application Registry and Events Service API, however, it is not recommeded to use these. You should call metadata endpoint URL, which is provided in `infoUrl` property, to fetch correct URLs to Application Registry and Events API, and other configuration details.
```
Comment (mmitoraj) [inline docs/application-connector/docs/08-02-get-client-certificate.md]:
```suggestion
> **NOTE:** The response contains URLs to the Application Registry and the Events Service API, however, it is not recommended to use them. Call the `metadata` endpoint URL, provided in the `infoUrl` property, to fetch the correct URLs and other configuration details.
```

File: docs/application-connector/docs/08-02-get-client-certificate.md [hunk 3]
```diff
@@ -100,15 +102,49 @@ The response contains a valid client certificate signed by the Kyma Certificate
 
 After you receive the certificate, decode it and use it in your application. Register the services of your external solution through the Application Registry.
 
-## Call the Metadata and Event services on local deployment
+## Call metadata endpoint
```
Comment (mmitoraj) [inline docs/application-connector/docs/08-02-get-client-certificate.md]:
```suggestion
## Call the `metadata` endpoint
```

File: docs/application-connector/docs/08-02-get-client-certificate.md [hunk 4]
```diff
@@ -100,15 +102,49 @@ The response contains a valid client certificate signed by the Kyma Certificate
...
+
+You must call the metadata endpoint with the generated certificate to get URLs to the following:
```
Comment (mmitoraj) [inline docs/application-connector/docs/08-02-get-client-certificate.md]:
```suggestion
Call the `metadata` endpoint with the generated certificate to get URLs to the following:
```

File: docs/application-connector/docs/08-02-get-client-certificate.md [hunk 5]
```diff
@@ -100,15 +102,49 @@ The response contains a valid client certificate signed by the Kyma Certificate
...
+
+- Application Registry API
```
Comment (mmitoraj) [inline docs/application-connector/docs/08-02-get-client-certificate.md]:
```suggestion
- the Application Registry API
```

File: docs/application-connector/docs/08-02-get-client-certificate.md [hunk 6]
```diff
@@ -100,15 +102,49 @@ The response contains a valid client certificate signed by the Kyma Certificate
...
+- Events Service API
```
Comment (mmitoraj) [inline docs/application-connector/docs/08-02-get-client-certificate.md]:
```suggestion
- the Events Service API
```

File: docs/application-connector/docs/08-02-get-client-certificate.md [hunk 7]
```diff
@@ -100,15 +102,49 @@ The response contains a valid client certificate signed by the Kyma Certificate
...
+- Certificate renewal endpoint
```
Comment (mmitoraj) [inline docs/application-connector/docs/08-02-get-client-certificate.md]:
```suggestion
- the `certificate renewal` endpoint
```

File: docs/application-connector/docs/08-02-get-client-certificate.md [hunk 8]
```diff
@@ -100,15 +102,49 @@ The response contains a valid client certificate signed by the Kyma Certificate
...
+- Certificate revocation endpoint
```
Comment (mmitoraj) [inline docs/application-connector/docs/08-02-get-client-certificate.md]:
```suggestion
- the `certificate revocation` endpoint
```

File: docs/application-connector/docs/08-02-get-client-certificate.md [hunk 9]
```diff
@@ -100,15 +102,49 @@ The response contains a valid client certificate signed by the Kyma Certificate
...
+
+Use the link you got in the second step to fetch the metadata information. Run:
```
Comment (mmitoraj) [inline docs/application-connector/docs/08-02-get-client-certificate.md]: Maybe providing the exact link would be a better idea?

File: docs/application-connector/docs/08-02-get-client-certificate.md [hunk 10]
```diff
@@ -100,15 +102,49 @@ The response contains a valid client certificate signed by the Kyma Certificate
...
+```
+{
+  "clientIdentity": {
+    "application": "APP_NAME"
+  },
+  "urls": {
+    "metadataUrl": "https://gateway.{CLUSTER_DOMAIN}/{APP_NAME}/v1/metadata/services",
+    "eventsUrl": "https://gateway.{CLUSTER_DOMAIN}/{APP_NAME}/v1/events",
+    "renewCertUrl": "https://gateway.{CLUSTER_DOMAIN}/v1/applications/certificates/renewals",
+    "revokeCertUrl": "https://gateway.{CLUSTER_DOMAIN}/v1/applications/certificates/revocations"
```
[the remark this answers is not in the document]
Comment (akgalwas) [reply docs/application-connector/docs/08-02-get-client-certificate.md]: Done

File: docs/application-connector/docs/08-02-get-client-certificate.md [hunk 11]
```diff
@@ -100,15 +102,49 @@ The response contains a valid client certificate signed by the Kyma Certificate
...
+Use the link you got in the second step to fetch the metadata information. Run:
+
+```
+curl {METADATA_ENDPOINT_URL} --cert {CERT_FILE_NAME}.crt --key {KEY_FILE_NAME}.key
+```
+
+```
+{
+  "clientIdentity": {
+    "application": "APP_NAME"
```
[the remark this answers is not in the document]
Comment (akgalwas) [reply docs/application-connector/docs/08-02-get-client-certificate.md]: Done

File: docs/application-connector/docs/02-02-connector-service.md (hunk 1 above)
Comment (akgalwas) [reply docs/application-connector/docs/02-02-connector-service.md]: Done

File: docs/application-connector/docs/02-02-connector-service.md [hunk 12]
```diff
@@ -13,8 +13,19 @@ This diagram illustrates the client certificate generation flow in details:
 1. The administrator requests for a token using the CLI or the UI and receives a link with the token, which is valid for a limited period of time.
 2. The administrator passes the token to the external system, which requests for information regarding the Kyma installation. In the response, it receives the following information:
     - the URL to which a third-party solution sends its Certificate Signing Request (CSR)
-    - URLs of the available APIs
+    - the URL of the metadata endpoint
```
[the remark this answers is not in the document]
Comment (akgalwas) [reply docs/application-connector/docs/02-02-connector-service.md]: Done

File: docs/application-connector/docs/02-02-connector-service.md [hunk 13]
```diff
@@ -13,8 +13,19 @@ This diagram illustrates the client certificate generation flow in details:
...
     - information required to generate a CSR
 3. The external system generates a CSR based on the information provided by Kyma and sends the CSR to the designated URL. In the response, the external system receives a signed certificate. It can use the certificate to authenticate and safely communicate with Kyma.
+4. The external system calls the metadata endpoint that contains the following information:
```
[the remark this answers is not in the document]
Comment (akgalwas) [reply docs/application-connector/docs/02-02-connector-service.md]: Done

File: docs/application-connector/docs/02-02-connector-service.md [hunk 14]
```diff
@@ -13,8 +13,19 @@ This diagram illustrates the client certificate generation flow in details:
...
+    - the URL of the Application Registry API
+    - the URL of the Event Service API
+    - the certificate renewal URL that should be used to rotate certificate
```
[the remark this answers is not in the document]
Comment (akgalwas) [reply docs/application-connector/docs/02-02-connector-service.md]: Done

File: docs/application-connector/docs/02-02-connector-service.md [hunk 15]
```diff
@@ -13,8 +13,19 @@ This diagram illustrates the client certificate generation flow in details:
...
+    - the certificate revocation URL that should be used to revoke compromised certificates
```
[the remark this answers is not in the document]
Comment (akgalwas) [reply docs/application-connector/docs/02-02-connector-service.md]: Done

File: docs/application-connector/docs/02-02-connector-service.md [hunk 16]
```diff
@@ -13,8 +13,19 @@ This diagram illustrates the client certificate generation flow in details:
...
+    - information uniquely identyfying the certificate (Application name)
```
[the remark this answers is not in the document]
Comment (akgalwas) [reply docs/application-connector/docs/02-02-connector-service.md]: Done

File: docs/application-connector/docs/08-02-get-client-certificate.md (hunk 5 above)
Comment (akgalwas) [reply docs/application-connector/docs/08-02-get-client-certificate.md]: Done

File: docs/application-connector/docs/08-02-get-client-certificate.md (hunk 6 above)
Comment (akgalwas) [reply docs/application-connector/docs/08-02-get-client-certificate.md]: Done

File: docs/application-connector/docs/08-02-get-client-certificate.md (hunk 7 above)
Comment (akgalwas) [reply docs/application-connector/docs/08-02-get-client-certificate.md]: Done

File: docs/application-connector/docs/08-02-get-client-certificate.md (hunk 8 above)
Comment (akgalwas) [reply docs/application-connector/docs/08-02-get-client-certificate.md]: Done

File: docs/application-connector/docs/08-02-get-client-certificate.md (hunk 9 above)
Comment (akgalwas) [reply docs/application-connector/docs/08-02-get-client-certificate.md]: Done

File: docs/application-connector/docs/02-02-connector-service.md [hunk 17]
```diff
@@ -13,8 +13,19 @@ This diagram illustrates the client certificate generation flow in details:
...
+    - information required to generate a CSR  
+
+>**NOTE:** The external application should not hardcode any URLs. The information returned from metadata endpoint should be stored by the external application along with the certificate. This approach implicates less coupling and offers a great deal of flexibility.
```
[the remark this answers is not in the document]
Comment (mmitoraj) [reply docs/application-connector/docs/02-02-connector-service.md]: Could you also apply this suggestion?

File: docs/application-connector/docs/02-02-connector-service.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (mmitoraj) [reply docs/application-connector/docs/02-02-connector-service.md]: Could you also apply this suggestion?

File: docs/application-connector/docs/08-02-get-client-certificate.md (hunk 3 above)
Comment (mmitoraj) [reply docs/application-connector/docs/08-02-get-client-certificate.md]: Could you also apply this suggestion?

File: docs/application-connector/docs/08-02-get-client-certificate.md (hunk 4 above)
Comment (mmitoraj) [reply docs/application-connector/docs/08-02-get-client-certificate.md]: Could you also apply this suggestion?

File: docs/application-connector/docs/08-02-get-client-certificate.md [hunk 18]
```diff
@@ -75,6 +75,8 @@ A successful call returns the following response:
 }
 ```
 
+> **NOTE:** The response contains URLs to the Application Registry and Events Service API, however, it is not recommeded to use these. You should call the `metadata` endpoint URL, which is provided in `infoUrl` property, to fetch correct URLs to the Application Registry and Events API, and other configuration details.
```
Comment (mmitoraj) [inline docs/application-connector/docs/08-02-get-client-certificate.md]:
```suggestion
> **NOTE:** The response contains URLs to the Application Registry and the Events Service API, however, it is not recommended to use them. You should call the `metadata` endpoint URL, which is provided in `infoUrl` property, to fetch correct URLs to the Application Registry and to the Events API, and other configuration details.
```

File: docs/application-connector/docs/08-02-get-client-certificate.md [hunk 19]
```diff
@@ -100,15 +102,49 @@ The response contains a valid client certificate signed by the Kyma Certificate
 
 After you receive the certificate, decode it and use it in your application. Register the services of your external solution through the Application Registry.
 
-## Call the Metadata and Event services on local deployment
+## Call the `metadata` endpoint
```
Comment (mmitoraj) [inline docs/application-connector/docs/08-02-get-client-certificate.md]:
```suggestion
## Call the metadata endpoint
```

File: docs/application-connector/docs/08-02-get-client-certificate.md [hunk 20]
```diff
@@ -100,15 +102,49 @@ The response contains a valid client certificate signed by the Kyma Certificate
...
+
+You must call the `metadata` endpoint with the generated certificate to get URLs to the following:
```
Comment (mmitoraj) [inline docs/application-connector/docs/08-02-get-client-certificate.md]:
```suggestion
Call the `metadata` endpoint with the generated certificate to get URLs to the following:
```

File: docs/application-connector/docs/08-02-get-client-certificate.md [hunk 21]
```diff
@@ -100,15 +102,49 @@ The response contains a valid client certificate signed by the Kyma Certificate
...
+
+- the Application Registry API
+- the Events Service API
+- the `certificate renewal` endpoint
+- the `certificate revocation` endpoint
+
+Use the link you got in the second step to fetch the metadata information. Run:
```
Comment (mmitoraj) [inline docs/application-connector/docs/08-02-get-client-certificate.md]:
```suggestion
Use the link provided in the second step to fetch the metadata information. Run:
```
Do you mean this link `https://connector-service.kyma.local/v1/applications/signingRequests/info?token=h31IwJiLNjnbqIwTPnzLuNmFYsCZeUtVbUvYL2hVNh6kOqFlW9zkHnzxYFCpCExBZ_voGzUo6IVS_ExlZd4muQ==`?

File: docs/application-connector/docs/08-02-get-client-certificate.md [hunk 22]
```diff
@@ -100,15 +102,49 @@ The response contains a valid client certificate signed by the Kyma Certificate
...
+
+```
+curl {CLUSTER_DOMAIN}/v1/applications/management/info --cert {CERT_FILE_NAME}.crt --key {KEY_FILE_NAME}.key
+```
+
```
Comment (mmitoraj) [inline docs/application-connector/docs/08-02-get-client-certificate.md]: What is it in the code blocks?

File: docs/application-connector/docs/08-02-get-client-certificate.md (hunk 18 above)
Comment (akgalwas) [reply docs/application-connector/docs/08-02-get-client-certificate.md]: Done

File: docs/application-connector/docs/08-02-get-client-certificate.md (hunk 19 above)
Comment (akgalwas) [reply docs/application-connector/docs/08-02-get-client-certificate.md]: Done

File: docs/application-connector/docs/08-02-get-client-certificate.md (hunk 20 above)
Comment (akgalwas) [reply docs/application-connector/docs/08-02-get-client-certificate.md]: Done

File: docs/application-connector/docs/08-02-get-client-certificate.md (hunk 21 above)
Comment (akgalwas) [reply docs/application-connector/docs/08-02-get-client-certificate.md]: Done

File: docs/application-connector/docs/08-02-get-client-certificate.md (hunk 22 above)
Comment (akgalwas) [reply docs/application-connector/docs/08-02-get-client-certificate.md]: Done.

**Pull request merged**